datalad/distribution/tests/test_install.py

```diff
@@ -270,6 +271,20 @@ def test_install_dataset_from_just_source(url, path):
     assert_in('INFO.txt', ds.repo.get_indexed_files())
 
 
+@with_testrepos(flavors=['local'])
+@with_tempfile(mkdir=True)
+def test_install_dataset_from_instance(src, dst):
+    origin = Dataset(src)
+    clone = install(source=origin, path=dst)
+
+    assert_is_instance(clone, Dataset)
+    ok_startswith(clone.path, dst)
+    ok_(clone.is_installed())
+    ok_(GitRepo.is_valid_repo(clone.path))
+    ok_clean_git(clone.path, annex=None)
+    assert_in('INFO.txt', clone.repo.get_indexed_files())
+
```

Comment:
With those tests we can REALLY be sure! Lines 280 + 284-5 should be more than sufficient IMHO ;-)